client/app/components/users/UserEdit.jsx

```diff
@@ -0,0 +1,279 @@
+
+    this.setState({ togglingUser: true });
+    toggleUser(user).then(({ data }) => {
+      this.setState({ user: User.convertUserInfo(data) });
+    }).finally(() => {
+      this.setState({ togglingUser: false });
+    });
+  };
+
+  renderBasicInfoForm() {
+    const { user } = this.state;
+    const formFields = [
+      {
+        name: 'name',
+        title: 'Name',
+        type: 'text',
+        initialValue: user.name,
+        required: true,
+      },
+      {
+        name: 'email',
+        title: 'Email',
+        type: 'email',
+        initialValue: user.email,
+        required: true,
+      },
+    ];
+
+    return (
+      <DynamicForm
+        fields={formFields}
+        readOnly={user.isDisabled}
+        onSubmit={this.onSaveUser}
+      />
+    );
+  }
+
+  renderChangePassword() {
+    const fields = [
+      { name: 'old_password', title: 'Current Password' },
+      { name: 'password', title: 'New Password', minLength: 6 },
+      { name: 'password_repeat', title: 'Repeat New Password' },
+    ].map(field => ({ ...field, type: 'password', required: true }));
+
+    return (
+      <Fragment>
+        <hr />
+        <Modal
+          visible={this.state.changingPassword}
+          title="Change Password"
+          onCancel={() => { this.setState({ changingPassword: false }); }}
+          footer={null}
+          destroyOnClose
+        >
+          <DynamicForm fields={fields} saveText="Update Password" onSubmit={this.onUpdatePassword} />
+        </Modal>
+        <Button className="w-100 m-t-10" onClick={this.onClickChangePassword} data-test="ChangePassword">
+          Change Password
+        </Button>
+      </Fragment>
+    );
+  }
+
+  renderApiKey() {
+    const { user } = this.state;
+
+    const regenerateButton = (
+      <Tooltip title="Regenerate API Key">
+        <Icon
+          type="reload"
+          style={{ cursor: 'pointer' }}
+          data-test="RegenerateApiKey"
+          onClick={this.onClickRegenerateApiKey}
+        />
+      </Tooltip>
+    );
+
+    return (
+      <Form layout="vertical">
+        <hr />
+        <Form.Item label="API Key">
+          <Input id="apiKey" addonAfter={regenerateButton} value={user.apiKey} data-test="ApiKey" readOnly />
+        </Form.Item>
+      </Form>
+    );
+  }
+
+  renderPasswordLinkAlert() {
+    const { user, passwordResetLink } = this.state;
+
+    return (
+      <Alert
+        message={clientConfig.mailSettingsMissing ? (
+          <p>
+            The mail server is not configured, please send the following link
+            to {user.name} to reset their password:
+            <Input.TextArea value={absoluteUrl(passwordResetLink)} readOnly />
+          </p>
+        ) : 'The user should receive a link to reset their password by email soon.'}
+        type="success"
+        className="m-t-20"
+        afterClose={() => { this.setState({ passwordResetLink: null }); }}
+        closable
+      />
```

Comment:
Wdyt about having this in a Modal or Popover instead?

Reply:
> Toastr with a shorter message for when the email server is configured

Definitely 👍 

> both a modal or a popover seem unexpected to me when I click on that button

I agree that the popover is irrelevant.
But I also think the current alert doesn't communicate the right message to the user.

We need to communicate:
1) Email not sent!
2) The reason is misconfiguration

And offer the user:
1) To fix configuration -  link to the relevant setting, perhaps a dedicated knowledge base help page.
2) Alternative by sending the link manually. With a [copy button](https://codesandbox.io/s/onxv7zxy5) perhaps.

Reply:
I made some changes on the existing `Alert` to make it communicate better:

![send-password-reset-email](https://user-images.githubusercontent.com/3356951/52526003-79682e80-2c99-11e9-9e74-620e77ced75a.gif)

Some notes:
- I took the opportunity to create an `InputWithCopy` component (I thought at first with `Input` instead of `<code>`, but LMK if you have another opinion).
- The page I found for related settings was [this](https://redash.io/help/open-source/setup#Mail-Configuration). My only concern is that this page will probably not be relevant to most Admins that will see this Alert, so I would not put this as the first option. I'll try to think somewhere to put it, though I think it could be interesting to add it to the existing Settings Screen -> Email Settings Warning.
- BTW, I'll check how it looks with a Modal instead of an Alert

Reply:
>  `Input` instead of `<code>`

It's perfect.

> this page will probably not be relevant to most Admins

How so?
@arikfr I think your input could be valuable here.

> BTW, I'll check how it looks with a Modal instead of an Alert

I think it's good enough the way it is. Let's move on :)

Reply:
I agree that the last version you show in the gif looks perfect 👌 

> * My only concern is that this page will probably not be relevant to most Admins that will see this Alert,

By the assumption that not all admins manage the server itself?

Reply:
Btw, <InputWithCopy> is useful for the API key too :)

Reply:
Yeah, perhaps copy button where the regenerate button is now, and move regenerate button  underneath.

(Also, add title "Password" above the two password buttons)

Reply:
> By the assumption that not all admins manage the server itself?

Yes, not all admins will have access to environment stuff.

> Yeah, perhaps copy button where the regenerate button is now, and move regenerate button underneath.

I had that even written on the description before, I guess I was just waiting for another use for a `InputWithCopy` component to do it. Let's try this thing :)

Reply:
**Overall looking**
![overall-apikey](https://user-images.githubusercontent.com/3356951/52535436-699e2800-2d35-11e9-9557-7e4b2d604536.gif)

**Pending Invitation view**
![pending-invitation](https://user-images.githubusercontent.com/3356951/52535449-92262200-2d35-11e9-94b7-0547acbfdf2e.png)

I was in doubt if I would keep Resend Invitation here, but then I was thinking: does it make sense to have a Change Password for Pending Invitation users?

Reply:
The way I see it:
Change password - user.id === currentUser.id
Resend invitation - currentUser.isAdmin && user.isInvitationPending
Reset password - currentUser.isAdmin && user.id !== currentUser.id